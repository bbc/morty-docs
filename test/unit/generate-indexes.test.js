const generateIndexes = require('../../src/generate-indexes.js')

describe('Generate Indexes with the correct paths', () => {
  it('for a top level index page', () => {
    const input = [{ relativePath: 'someRootFile.html' }]

    const actual = generateIndexes(input, { repoName: 'some-repo' })

    const expected = [{
      relativePath: 'index.html',
      raw: expect.anything()
    }]

    expect(actual).toEqual(expected)
  })

  it('for multiple index pages', () => {
    const input = [{
      relativePath: 'someRootFile.html'
    }, {
      relativePath: 'someFolder/file.html'
    }, {
      relativePath: 'someOtherFolder/file.html'
    }]

    const actual = generateIndexes(input, { repoName: 'some-repo' })

    const expected = [{
      relativePath: 'index.html',
      raw: expect.anything()
    }, {
      relativePath: 'someFolder/index.html',
      raw: expect.anything()
    }, {
      relativePath: 'someOtherFolder/index.html',
      raw: expect.anything()
    }]

    expect(actual).toEqual(expected)
  })

  it('generates indexes for every directory in path', () => {
    const input = [{
      relativePath: 'someFolder/someOtherFolder/docs/file.html'
    }]

    const actual = generateIndexes(input, { repoName: 'some-repo' })

    const expected = [{
      relativePath: 'someFolder/index.html',
      raw: expect.anything()
    }, {
      relativePath: 'someFolder/someOtherFolder/index.html',
      raw: expect.anything()
    }, {
      relativePath: 'someFolder/someOtherFolder/docs/index.html',
      raw: expect.anything()
    }]

    expect(actual).toEqual(expected)
  })
})

describe('Generate Indexes with the correct html', () => {
  it('The correct links are included in the raw output', () => {
    const input = [{
      relativePath: 'rootFile.html'
    }, {
      relativePath: 'folder/nestedFile.html'
    }, {
      relativePath: 'docs/arch/furtherNestedFile.html'
    }]

    const actual = generateIndexes(input, { repoName: 'some-repo' })

    expect(actual[0].raw.toString()).toMatchSnapshot()
    expect(actual[1].raw.toString()).toMatchSnapshot()
    expect(actual[2].raw.toString()).toMatchSnapshot()
    expect(actual[3].raw.toString()).toMatchSnapshot()
  })

  it('to match snapshot with a content title', () => {
    const input = [{
      relativePath: 'foo1.html'
    }, {
      relativePath: 'foo2.html'
    }]

    const actual = generateIndexes(input, { contentTitle: 'some-repo' })

    expect(actual[0].raw.toString()).toMatchSnapshot()
  })

  it('to match snapshot without a content title', () => {
    const input = [{
      relativePath: 'foo1.html'
    }, {
      relativePath: 'foo2.html'
    }]

    const actual = generateIndexes(input, { })

    expect(actual[0].raw.toString()).toMatchSnapshot()
  })
})
