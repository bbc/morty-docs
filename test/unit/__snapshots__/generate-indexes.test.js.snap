// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate Indexes with the correct html The correct links are included in the raw output 1`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      Morty Docs
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
    <link rel="stylesheet"
          href="https://use.fontawesome.com/releases/v5.8.1/css/all.css"
          integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;min-height:60px;margin-bottom:0;width:100%;padding-right:5px;padding-left:5px"
      >
        <div class="row col-md-12"
             style="width:100%;padding-top:10px;padding-right:0px;padding-left:25px"
        >
          <h3 class="col-md-4"
              style="text-align:left;padding-right:0px;padding-bottom:20px;color:#fff"
          >
            Morty-Docs
          </h3>
        </div>
      </div>
      <div class="container"
           style="margin-top:10px"
      >
        <div class="row col-md-8 col-md-offset-2">
          <ul class="list-unstyled">
            <li class="index-list"
                style="font-size:1.6em;border-bottom:solid black 1px;margin-bottom:10px"
            >
              <a href="folder/index.html">
                <i class="fas fa-folder-open">
                </i>
                <span style="margin-left:5px">
                  folder
                </span>
              </a>
            </li>
            <li class="index-list"
                style="font-size:1.6em;border-bottom:solid black 1px;margin-bottom:10px"
            >
              <a href="docs/index.html">
                <i class="fas fa-folder-open">
                </i>
                <span style="margin-left:5px">
                  docs
                </span>
              </a>
            </li>
            <li class="index-list"
                style="font-size:1.6em;border-bottom:solid black 1px;margin-bottom:10px"
            >
              <a href="rootFile.html">
                <i class="fas fa-file-alt">
                </i>
                <span style="margin-left:5px">
                  rootFile.html
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;

exports[`Generate Indexes with the correct html The correct links are included in the raw output 2`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      Morty Docs
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
    <link rel="stylesheet"
          href="https://use.fontawesome.com/releases/v5.8.1/css/all.css"
          integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;min-height:60px;margin-bottom:0;width:100%;padding-right:5px;padding-left:5px"
      >
        <div class="row col-md-12"
             style="width:100%;padding-top:10px;padding-right:0px;padding-left:25px"
        >
          <h3 class="col-md-4"
              style="text-align:left;padding-right:0px;padding-bottom:20px;color:#fff"
          >
            Morty-Docs
          </h3>
        </div>
      </div>
      <div class="container"
           style="margin-top:10px"
      >
        <div class="row col-md-8 col-md-offset-2">
          <ul class="list-unstyled">
            <li class="index-list"
                style="font-size:1.6em;border-bottom:solid black 1px;margin-bottom:10px"
            >
              <a href="nestedFile.html">
                <i class="fas fa-file-alt">
                </i>
                <span style="margin-left:5px">
                  nestedFile.html
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;

exports[`Generate Indexes with the correct html The correct links are included in the raw output 3`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      Morty Docs
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
    <link rel="stylesheet"
          href="https://use.fontawesome.com/releases/v5.8.1/css/all.css"
          integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;min-height:60px;margin-bottom:0;width:100%;padding-right:5px;padding-left:5px"
      >
        <div class="row col-md-12"
             style="width:100%;padding-top:10px;padding-right:0px;padding-left:25px"
        >
          <h3 class="col-md-4"
              style="text-align:left;padding-right:0px;padding-bottom:20px;color:#fff"
          >
            Morty-Docs
          </h3>
        </div>
      </div>
      <div class="container"
           style="margin-top:10px"
      >
        <div class="row col-md-8 col-md-offset-2">
          <ul class="list-unstyled">
            <li class="index-list"
                style="font-size:1.6em;border-bottom:solid black 1px;margin-bottom:10px"
            >
              <a href="arch/index.html">
                <i class="fas fa-folder-open">
                </i>
                <span style="margin-left:5px">
                  arch
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;

exports[`Generate Indexes with the correct html The correct links are included in the raw output 4`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      Morty Docs
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
    <link rel="stylesheet"
          href="https://use.fontawesome.com/releases/v5.8.1/css/all.css"
          integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;min-height:60px;margin-bottom:0;width:100%;padding-right:5px;padding-left:5px"
      >
        <div class="row col-md-12"
             style="width:100%;padding-top:10px;padding-right:0px;padding-left:25px"
        >
          <h3 class="col-md-4"
              style="text-align:left;padding-right:0px;padding-bottom:20px;color:#fff"
          >
            Morty-Docs
          </h3>
        </div>
      </div>
      <div class="container"
           style="margin-top:10px"
      >
        <div class="row col-md-8 col-md-offset-2">
          <ul class="list-unstyled">
            <li class="index-list"
                style="font-size:1.6em;border-bottom:solid black 1px;margin-bottom:10px"
            >
              <a href="furtherNestedFile.html">
                <i class="fas fa-file-alt">
                </i>
                <span style="margin-left:5px">
                  furtherNestedFile.html
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;

exports[`Generate Indexes with the correct html to match snapshot with a content title 1`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      Morty Docs
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
    <link rel="stylesheet"
          href="https://use.fontawesome.com/releases/v5.8.1/css/all.css"
          integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;min-height:60px;margin-bottom:0;width:100%;padding-right:5px;padding-left:5px"
      >
        <div class="row col-md-12"
             style="width:100%;padding-top:10px;padding-right:0px;padding-left:25px"
        >
          <h3 class="col-md-4"
              style="text-align:left;padding-right:0px;padding-bottom:20px;color:#fff"
          >
            Morty-Docs
          </h3>
        </div>
      </div>
      <div class="container"
           style="margin-top:10px"
      >
        <div class="row col-md-6 col-md-offset-3"
             style="text-align:center;margin-bottom:20px"
        >
          <h1 style="font-size:32px">
            some-repo
          </h1>
        </div>
        <div class="row col-md-8 col-md-offset-2">
          <ul class="list-unstyled">
            <li class="index-list"
                style="font-size:1.6em;border-bottom:solid black 1px;margin-bottom:10px"
            >
              <a href="foo1.html">
                <i class="fas fa-file-alt">
                </i>
                <span style="margin-left:5px">
                  foo1.html
                </span>
              </a>
            </li>
            <li class="index-list"
                style="font-size:1.6em;border-bottom:solid black 1px;margin-bottom:10px"
            >
              <a href="foo2.html">
                <i class="fas fa-file-alt">
                </i>
                <span style="margin-left:5px">
                  foo2.html
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;

exports[`Generate Indexes with the correct html to match snapshot without a content title 1`] = `
<html lang="en"
      style="min-height:100vh"
      data-reactroot
>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible"
          content="IE=edge"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <title>
      Morty Docs
    </title>
    <link rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
          integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
          crossorigin="anonymous"
    >
    <link rel="stylesheet"
          href="https://use.fontawesome.com/releases/v5.8.1/css/all.css"
          integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf"
          crossorigin="anonymous"
    >
  </head>
  <body style="min-height:100vh">
    <div style="min-height:75vh;padding-bottom:20px">
      <div class="container"
           style="border:none;border-radius:0;background-color:#000;min-height:60px;margin-bottom:0;width:100%;padding-right:5px;padding-left:5px"
      >
        <div class="row col-md-12"
             style="width:100%;padding-top:10px;padding-right:0px;padding-left:25px"
        >
          <h3 class="col-md-4"
              style="text-align:left;padding-right:0px;padding-bottom:20px;color:#fff"
          >
            Morty-Docs
          </h3>
        </div>
      </div>
      <div class="container"
           style="margin-top:10px"
      >
        <div class="row col-md-8 col-md-offset-2">
          <ul class="list-unstyled">
            <li class="index-list"
                style="font-size:1.6em;border-bottom:solid black 1px;margin-bottom:10px"
            >
              <a href="foo1.html">
                <i class="fas fa-file-alt">
                </i>
                <span style="margin-left:5px">
                  foo1.html
                </span>
              </a>
            </li>
            <li class="index-list"
                style="font-size:1.6em;border-bottom:solid black 1px;margin-bottom:10px"
            >
              <a href="foo2.html">
                <i class="fas fa-file-alt">
                </i>
                <span style="margin-left:5px">
                  foo2.html
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <footer style="bottom:0;width:100%;background-color:#f5f5f5;padding:0 15px;box-sizing:border-box;min-height:25vh;position:relative"
            class="footer"
    >
      <div style="padding:1em 0 2em 0"
           class="container"
      >
        <div class="row">
          <div style="text-align:center"
               class="col-md-4 col-md-offset-4"
          >
            <h3>
              What is Morty-Docs?
            </h3>
            <p>
              Link to
              <a href="https://github.com/bbc/morty-docs">
                Morty-docs
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </footer>
  </body>
</html>
`;
